### PR TITLE
User story 47/48 pd

### LINKED ISSUE #78: User Story 47, Merchant edits an item
As a merchant
When I visit my items page
And I click the edit button or link next to any item
Then I am taken to a form similar to the 'new item' form
The form is pre-populated with all of this item's information
I can change any information, but all of the rules for adding a new item still apply:
- name and description cannot be blank
- price cannot be less than $0.00
- inventory must be 0 or greater

When I submit the form
I am taken back to my items page
I see a flash message indicating my item is updated
I see the item's new information on the page, and it maintains its previous enabled/disabled state
If I left the image field blank, I see a placeholder image for the thumbnail

### PR DESCRIPTION
**Pull request by philjdelong**
## Description
Resolves #78, #79 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## RSpec Results
```
Finished in 10.48 seconds (files took 2.29 seconds to load)
221 examples, 0 failures


Coverage report generated for RSpec to /Users/phildelong/turing/mod2/projects/monster_shop_part_1/coverage. 2681
 / 2682 LOC (99.96%) covered.
```

### CODE REVIEW

File: app/controllers/merchant/items_controller.rb [hunk 1]
```diff
@@ -4,7 +4,23 @@ def index
...
 
+  def edit
+    @item = Item.find(params[:id])
+  end
+
   def update
+    item = Item.find(params[:id])
+    item.update(item_params)
+    if item.save
+      flash[:success] = "Yer booty has been corrected!"
```
Comment (DanielEFrampton) [inline app/controllers/merchant/items_controller.rb:30]: Yeah that got a lol out of me

**Review by DanielEFrampton — APPROVED** [review]
Nice work!

**Review by mel-rob — APPROVED** [review]
Looks great! Thanks!

**Review by DanielEFrampton — APPROVED** [review]
Looks gud

**Review by KirbyDD — APPROVED** [review]

**Pull request merged**